### PR TITLE
Feat/add v1.24.7

### LINKED ISSUE #20: Add support for kubernetes 1.24
As per title

### PR DESCRIPTION
**Pull request by nutellinoit**
This PR adds the support for Kubernetes 1.24.7

- [x] Update the ansible roles to include 1.24.7
  - [x] Test Rocky 8
  - [x] Test Ubuntu 20
- [x] Upgrade containerd to 1.6.8 (required 1.6.4+ by Kubernetes 1.24)
- [x] Update vmware components
  - [x] controller manager
  - [x] CSI 
- [x] Write Release notes

Fixes #20

### CODE REVIEW

File: docs/releases/v1.24.7.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+## Update Guide 🦮
+
+In this guide, we will try to summarize the update process from `v1.23.12` to this release.
+
+> NOTE: Each on-premises environment can be different, always double-check before updating components.
+
+1. Update the vSphere Controller Manager package and vSphere CSI if applicable (see below)
+2. Update containerd to the new version (see the [example playbooks](examples/playbooks))
+3. Update Kubernetes masters (see the [example playbooks](examples/playbooks))
```
Comment (ralgozino) [inline docs/releases/v1.24.7.md:29]:
```suggestion
3. Update Kubernetes control plane nodes (see the [example playbooks](examples/playbooks))
```

File: examples/playbooks/hosts.ini [hunk 2]
```diff
@@ -27,7 +27,7 @@ ansible_user=root
 ansible_ssh_private_key_file='../ssh-key'
 ansible_python_interpreter=python3
 kubernetes_kubeconfig_path='./'
-kubernetes_version='1.20.15'
+kubernetes_version='1.23.12'
```
Comment (ralgozino) [inline examples/playbooks/hosts.ini:30]:
```suggestion
kubernetes_version='1.24.7'
```

File: katalog/vsphere-cm/README.md [hunk 3]
```diff
@@ -4,7 +4,7 @@ This katalog deploys the [vSphere Cloud Controller Manager](https://github.com/k
 
 ## Requirements
 
-- Kubernetes = `1.23.x`
+- Kubernetes = `1.24.x`
 - Kustomize >= `v3.5.3`
 - control plane and the nodes must be provisioned with cloud-provider `external` on `kubeadm.yaml`
```
Comment (ralgozino) [inline katalog/vsphere-cm/README.md:9]:
```suggestion
- control plane and worker nodes must be provisioned with cloud-provider `external` on `kubeadm.yaml`
```

File: README.md [hunk 4]
```diff
@@ -40,7 +40,7 @@ Click on each package to see its full documentation.
 
 ## Compatibility
 
-This version is compatible with Kubernetes 1.23.12
+This version is compatible with Kubernetes 1.24.7
 
 Check the [compatibility matrix][compatibility-matrix] for additional information about previous releases of the modules.
```
Comment (ralgozino) [inline README.md:45]:
```suggestion
Check the [compatibility matrix][compatibility-matrix] for additional information about previous releases of the module.
```

File: docs/releases/v1.24.7.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+Welcome to the latest release of `on-premises` module of [`Kubernetes Fury Distribution`](https://github.com/sighupio/fury-distribution) maintained by SIGHUP team.
+
+This minor release adds the compatibility with Kubernetes 1.24.7.
+
+## Package Versions 🚢
+
+| Package                                              | Supported Version | Previous Version  |
+|------------------------------------------------------|-------------------|-------------------|
+| [vmware-cm](katalog/vmware-cm)                       | `1.24.2`          | `Updated`         |
```
Comment (lnovara) [inline docs/releases/v1.24.7.md:11]: Shouldn't you add the previous version under the "Previous Version" column?

Also, I would add a link to the upstream project changelog in the "Supported Version" column.

File: docs/releases/v1.24.7.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+```yaml
+kustomize build <your-project-path-including-vmware-cm-as-base> | kubectl apply -f -
+```
+
+
+## vsphere-csi
+
+The vSphere CSI driver update can be executed before upgrading the cluster version to v1.24.7.
+The current version is compatible with Kubernetes 1.22.x to 1.24.x.
```
Comment (lnovara) [inline docs/releases/v1.24.7.md:48]:
```suggestion
The current version is compatible with Kubernetes from v1.22.x to v1.24.x.
```

File: examples/playbooks/hosts.ini (hunk 2 above)
Comment (lnovara) [inline examples/playbooks/hosts.ini:30]: Why not 1.24.something?

File: katalog/vsphere-cm/vsphere-cloud-controller-manager.yaml [hunk 7]
```diff
@@ -216,14 +216,17 @@ spec:
         tier: control-plane
     spec:
       nodeSelector:
-        node-role.kubernetes.io/master: ""
+        node-role.kubernetes.io/control-plane: ""
```
Comment (lnovara) [inline katalog/vsphere-cm/vsphere-cloud-controller-manager.yaml:219]: Do all the Kubernetes versions supported by this module, support this "new" label?

**Review by lnovara — APPROVED** [review]
Some minor comments.

Overall, SBAM! :shipit:

**Review by ralgozino — CHANGES_REQUESTED** [review]
some very minor suggestions, additionally the default value for `kubernetes_version` in the file should be updated:
https://github.com/sighupio/fury-kubernetes-on-premises/blob/73eed4c47532c39cb3ada72b9cbd1038cc03dd9f/roles/kube-control-plane/defaults/main.yml#L17

I'll create an issue to track that we should eventually align with upstream nomenclature and switch also from `master` to `control plane`.

Do you think that adding a warning about this change in the release notes?

File: examples/playbooks/hosts.ini (hunk 2 above)
Comment (nutellinoit) [reply examples/playbooks/hosts.ini:30]: In the example I also show the upgrade procedure

File: katalog/vsphere-cm/vsphere-cloud-controller-manager.yaml (hunk 7 above)
Comment (nutellinoit) [reply katalog/vsphere-cm/vsphere-cloud-controller-manager.yaml:219]: Yes, 1.23 and 1.24 have "control-plane" as node-role

File: examples/playbooks/hosts.ini (hunk 2 above)
Comment (nutellinoit) [reply examples/playbooks/hosts.ini:30]: In the example I also show the upgrade procedure

File: docs/releases/v1.24.7.md (hunk 5 above)
Comment (nutellinoit) [reply docs/releases/v1.24.7.md:11]: You are right, let me change that

File: roles/containerd/defaults/main.yml [hunk 8]
```diff
@@ -2,11 +2,11 @@
 # Package options.
 
 # Check supported Kubernetes versions here: https://containerd.io/releases/#kubernetes-support
-containerd_version: 1.5.8
+containerd_version: 1.6.8
 containerd_download_url: "https://github.com/containerd/containerd/releases/download/v{{ containerd_version }}/containerd-{{ containerd_version }}-linux-{{ image_arch }}.tar.gz"
 containerd_bin_dir: "/usr/local/bin"
 # runc version from here: https://github.com/containerd/containerd/blob/main/docs/RUNC.md -> https://github.com/containerd/containerd/blob/main/script/setup/runc-version
-runc_version: v1.0.3
+runc_version: v1.1.4
```
Comment (ralgozino) [inline roles/containerd/defaults/main.yml:9]: maybe we can do a matrix like for the kubernetes components for this two versions?

**Review by ralgozino — APPROVED** [review]
lgtm!

File: roles/containerd/defaults/main.yml (hunk 8 above)
Comment (nutellinoit) [reply roles/containerd/defaults/main.yml:9]: uhm, let me try

**Review by lnovara — APPROVED** [review]

File: roles/containerd/defaults/main.yml (hunk 8 above)
Comment (nutellinoit) [reply roles/containerd/defaults/main.yml]: Added the new feature :)

**Review by ralgozino — APPROVED** [review]
🚀

**Pull request merged**